third_party/xla/xla/service/gpu/ir_emitter.cc

```diff
@@ -263,14 +263,7 @@ void IrEmitter::BindFusionArguments(const HloInstruction* fusion,
 void IrEmitter::MaybeEmitFenceForAMDGPU(llvm::AtomicOrdering atomic_ordering,
                                         const char* sync_scope_id) {
   if (IsEmittingForAMDGPU() &&
-<<<<<<< HEAD
-      (ir_emitter_context_->rocm_compute_capability().gcn_arch_name().substr(0, 6) == "gfx90a" || 
-       ir_emitter_context_->rocm_compute_capability().gcn_arch_name().substr(0, 6) == "gfx940" || 
-       ir_emitter_context_->rocm_compute_capability().gcn_arch_name().substr(0, 6) == "gfx941" || 
-       ir_emitter_context_->rocm_compute_capability().gcn_arch_name().substr(0, 6) == "gfx942")) {
-=======
       ir_emitter_context_->rocm_compute_capability().fence_before_barrier()) {
```

Comment:
@wenchenvincent , @i-chaochen, @pemeliya Is this the proper way to check if the barrier is needed now?

Reply:
http://ml-ci.amd.com:21096/blue/rest/organizations/jenkins/pipelines/tensorflow/pipelines/ThirdParty-XLA/branches/PR-2336/runs/3/nodes/41/steps/45/log/?start=0

Yes, I think `@local_xla//xla/stream_executor/gpu:gpu_kernel_test_gpu` is for this one, espeically on GpuKernelTilingTest.ColReductionWithSmallDtype https://github.com/ROCmSoftwarePlatform/frameworks-internal/issues/5866

Reply:
yeah these are what I was looking for:
https://github.com/openxla/xla/pull/7277
https://github.com/openxla/xla/commit/2fce0d6c94ad1fdba5def0100ca5bd85b7397bd6

should be good here